<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>7: Limpieza de datos</title>
    <meta charset="utf-8" />
    <meta name="author" content="Ana BVA" />
    <script src="libs/header-attrs-2.7/header-attrs.js"></script>
    <link href="libs/remark-css-0.0.1/default.css" rel="stylesheet" />
    <link href="libs/remark-css-0.0.1/default-fonts.css" rel="stylesheet" />
    <link href="libs/remark-css-0.0.1/hygge.css" rel="stylesheet" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# 7: Limpieza de datos
### Ana BVA
### 24/2/2021

---





&lt;style type="text/css"&gt;
/* From https://github.com/yihui/xaringan/issues/147  */
.scroll-output {
  height: 80%;
  overflow-y: scroll;
}
/* https://stackoverflow.com/questions/50919104/horizontally-scrollable-output-on-xaringan-slides */
pre {
  max-width: 100%;
  overflow-x: scroll;
}

&lt;/style&gt;


## ¿Porqué es importante revisar los datos crudos?
--

.full-width[.content-box-blue[Para checar la caldad de los datos (*QC: quality conrol*)]]

--

- Los datos influencian los análisis posteriores

--

.pull-left[
&lt;img src="https://miro.medium.com/max/470/1*DVLpt3zCNaqeZOGNpgME7Q.png" width="200px" style="display: block; margin: auto;" /&gt;&lt;img src="https://i.pinimg.com/originals/30/66/ac/3066ac69ae68ac200ace0ca8fe3882c3.jpg" width="200px" style="display: block; margin: auto;" /&gt;
]

.pull-right[

&lt;img src="https://memegenerator.net/img/instances/68955295.jpg" width="300px" style="display: block; margin: auto;" /&gt;

]


---

## FastQC

Usaremos el programa de [fastQC](https://www.bioinformatics.babraham.ac.uk/projects/download.html#fastqc) para checar la calidad de los datos.

--

Podemos usar la interfaz gráfica o la línea de comandos, en este caso usaremos la línea de comandos.

&lt;img src="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc.png" width="400px" style="display: block; margin: auto;" /&gt;

---

### Pasos para installar FastQC:
--

1- Descargar el programa de su página web [fastQC](https://www.bioinformatics.babraham.ac.uk/projects/download.html#fastqc)

--

2- Hacer el archivo ejecutable


```bash
# chmod 755 /Applications/FastQC.app/Contents/MacOS/fastqc
ls -l /Applications/FastQC.app/Contents/MacOS/fastqc
```

```
## -rwxr-xr-x@ 1 user  admin  14998 Jan  8  2020 /Applications/FastQC.app/Contents/MacOS/fastqc
```

--

3- Crear una liga simbolica para ejecutar `fastqc` desde cualquier ubicación


```bash
ln -s /Applications/FastQC.app/Contents/MacOS/fastqc /usr/local/bin/fastqc
```

--
4- Correr `fastqc`


```bash
fastqc --help
```

```
## 
##             FastQC - A high throughput sequence QC analysis tool
## 
## SYNOPSIS
## 
## 	fastqc seqfile1 seqfile2 .. seqfileN
## 
##     fastqc [-o output dir] [--(no)extract] [-f fastq|bam|sam] 
##            [-c contaminant file] seqfile1 .. seqfileN
## 
## DESCRIPTION
## 
##     FastQC reads a set of sequence files and produces from each one a quality
##     control report consisting of a number of different modules, each one of 
##     which will help to identify a different potential type of problem in your
##     data.
##     
##     If no files to process are specified on the command line then the program
##     will start as an interactive graphical application.  If files are provided
##     on the command line then the program will run with no user interaction
##     required.  In this mode it is suitable for inclusion into a standardised
##     analysis pipeline.
##     
##     The options for the program as as follows:
##     
##     -h --help       Print this help file and exit
##     
##     -v --version    Print the version of the program and exit
##     
##     -o --outdir     Create all output files in the specified output directory.
##                     Please note that this directory must exist as the program
##                     will not create it.  If this option is not set then the 
##                     output file for each sequence file is created in the same
##                     directory as the sequence file which was processed.
##                     
##     --casava        Files come from raw casava output. Files in the same sample
##                     group (differing only by the group number) will be analysed
##                     as a set rather than individually. Sequences with the filter
##                     flag set in the header will be excluded from the analysis.
##                     Files must have the same names given to them by casava
##                     (including being gzipped and ending with .gz) otherwise they
##                     won't be grouped together correctly.
##                     
##     --nano          Files come from nanopore sequences and are in fast5 format. In
##                     this mode you can pass in directories to process and the program
##                     will take in all fast5 files within those directories and produce
##                     a single output file from the sequences found in all files.                    
##                     
##     --nofilter      If running with --casava then don't remove read flagged by
##                     casava as poor quality when performing the QC analysis.
##                    
##     --extract       If set then the zipped output file will be uncompressed in
##                     the same directory after it has been created.  By default
##                     this option will be set if fastqc is run in non-interactive
##                     mode.
##                     
##     -j --java       Provides the full path to the java binary you want to use to
##                     launch fastqc. If not supplied then java is assumed to be in
##                     your path.
##                    
##     --noextract     Do not uncompress the output file after creating it.  You
##                     should set this option if you do not wish to uncompress
##                     the output when running in non-interactive mode.
##                     
##     --nogroup       Disable grouping of bases for reads &gt;50bp. All reports will
##                     show data for every base in the read.  WARNING: Using this
##                     option will cause fastqc to crash and burn if you use it on
##                     really long reads, and your plots may end up a ridiculous size.
##                     You have been warned!
##                     
##     --min_length    Sets an artificial lower limit on the length of the sequence
##                     to be shown in the report.  As long as you set this to a value
##                     greater or equal to your longest read length then this will be
##                     the sequence length used to create your read groups.  This can
##                     be useful for making directly comaparable statistics from 
##                     datasets with somewhat variable read lengths.
##                     
##     -f --format     Bypasses the normal sequence file format detection and
##                     forces the program to use the specified format.  Valid
##                     formats are bam,sam,bam_mapped,sam_mapped and fastq
##                     
##     -t --threads    Specifies the number of files which can be processed
##                     simultaneously.  Each thread will be allocated 250MB of
##                     memory so you shouldn't run more threads than your
##                     available memory will cope with, and not more than
##                     6 threads on a 32 bit machine
##                   
##     -c              Specifies a non-default file which contains the list of
##     --contaminants  contaminants to screen overrepresented sequences against.
##                     The file must contain sets of named contaminants in the
##                     form name[tab]sequence.  Lines prefixed with a hash will
##                     be ignored.
## 
##     -a              Specifies a non-default file which contains the list of
##     --adapters      adapter sequences which will be explicity searched against
##                     the library. The file must contain sets of named adapters
##                     in the form name[tab]sequence.  Lines prefixed with a hash
##                     will be ignored.
##                     
##     -l              Specifies a non-default file which contains a set of criteria
##     --limits        which will be used to determine the warn/error limits for the
##                     various modules.  This file can also be used to selectively 
##                     remove some modules from the output all together.  The format
##                     needs to mirror the default limits.txt file found in the
##                     Configuration folder.
##                     
##    -k --kmers       Specifies the length of Kmer to look for in the Kmer content
##                     module. Specified Kmer length must be between 2 and 10. Default
##                     length is 7 if not specified.
##                     
##    -q --quiet       Supress all progress messages on stdout and only report errors.
##    
##    -d --dir         Selects a directory to be used for temporary files written when
##                     generating report images. Defaults to system temp directory if
##                     not specified.
##                     
## BUGS
## 
##     Any bugs in fastqc should be reported either to simon.andrews@babraham.ac.uk
##     or in www.bioinformatics.babraham.ac.uk/bugzilla/
##                    
## 
```


---

### Pasos para correr FastQC:

.scroll-output[

1- Checar los datos fastq

```bash
pwd
ls ../data/Archivos_fastq
```

```
## /Users/user/Documents/Doctorado/Courses/Taught/minicurso_abr_2021/slides
## s1_R1.fastq.gz
## s1_R2.fastq.gz
## s2_R1.fastq.gz
## s2_R2.fastq.gz
## s3_R1.fastq.gz
## s3_R2.fastq.gz
## s4_R1.fastq.gz
## s4_R2.fastq.gz
## s5_R1.fastq.gz
## s5_R2.fastq.gz
## s6_R1.fastq.gz
## s6_R2.fastq.gz
```

--

2- Crear un directorio para guardar el output


```bash
mkdir ../output/QC
```

--

3- Correr `FastQC`


```bash
# la opción -o es para elejir el directorio de salida
fastqc ../data/Archivos_fastq/*.fastq.gz -o ../output/QC/
```

--

4- Checar los archivos generados


```bash
ls ../output/QC
```

```
## multiqc_data
## multiqc_report.html
## s1_R1_fastqc.html
## s1_R1_fastqc.zip
## s1_R2_fastqc.html
## s1_R2_fastqc.zip
## s2_R1_fastqc.html
## s2_R1_fastqc.zip
## s2_R2_fastqc.html
## s2_R2_fastqc.zip
## s3_R1_fastqc.html
## s3_R1_fastqc.zip
## s3_R2_fastqc.html
## s3_R2_fastqc.zip
## s4_R1_fastqc.html
## s4_R1_fastqc.zip
## s4_R2_fastqc.html
## s4_R2_fastqc.zip
## s5_R1_fastqc.html
## s5_R1_fastqc.zip
## s5_R2_fastqc.html
## s5_R2_fastqc.zip
## s6_R1_fastqc.html
## s6_R1_fastqc.zip
## s6_R2_fastqc.html
## s6_R2_fastqc.zip
```



]



---

## Buena calidad

El eje x = longitud de la read, y =  calidad (phred score)

Los datos a continuación muestran reads de buena calidad (Verde = buena, amarilla = aceptable, roja = baja). 

&lt;img src="https://gwu-omics2019.readthedocs.io/en/latest/_images/fastqc_good.png" width="550px" style="display: block; margin: auto;" /&gt;


---

## Mala calidad

¿Hasta donde los datos son de buena calidad? ¿Estos datos los usarías?

&lt;img src="https://gwu-omics2019.readthedocs.io/en/latest/_images/fastqc_bad.png" width="650px" style="display: block; margin: auto;" /&gt;


---

## Ejercicio

¿Cómo se ven nuestros datos? 


---
## MultiQC

[MultiQC](https://multiqc.info/) te permite juntar los output en un solo archivo y poder visualizarlos mejor.

*Esta parte es opcional, pero si trabajas con RNA-seq te puede intersar.

.scroll-output[

1. La instalación se hace con conda, así que primero activamos un ambiente conda


```bash
conda activate base
pip install multiqc
```


2. Correr MultiQC


```bash
cd ../output/.
multiQC .
```

3. Checar el [output.html](file:///Users/user/Documents/Doctorado/Courses/Taught/minicurso_abr_2021/output/QC/multiqc_report.html#fastqc_per_base_sequence_content)

]



    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script src="macros.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
